resources/js/app.js

```diff
@@ -11,24 +11,64 @@
 		}
 	})
 
-	app.directive('projects',  ['$http',  '$timeout' ,function ($http, $timeout) {
-		return {
-			restrict: 'E',
-			templateUrl: '/partials/tabs/projects.html',
-			controller: function ($scope) {
-				self = this
-				$scope.projectList = projects
+	app.directive('projects',  ['$http', '$timeout' ,'$location', 
+		function ($http, $timeout, $location) {
+			return {
+				restrict: 'E',
+				templateUrl: '/partials/tabs/projects.html',
+				controller: function ($scope) {
+					self = this
+
+					$scope.projectList = projects
+					$scope.projects_url_dict = {}
+
+					angular.forEach($scope.projectList, function(value, key){
+						value["url"] = value["name"].split(' ').join('_').toLowerCase();
+						$scope.projects_url_dict[value["url"]] = key
+					});
+
+				/* When project details view modal is closed
+				this function is called from modal callback 
+				from projects.html L100
+				*/
+				$scope.updateLink = function () {
+					$location.path('');
+				}
+
+				/* This sets the clicked project as
+				selected project, updates the url in
+				browser and opens the modal */
 				self.showProject = function (project) {
 					$(document).ready(function () {
-						$('#modal1').modal('open');
+						mval = project["name"].split(' ').join('_').toLowerCase();
+						$location.path('/' + mval)
+						$('.modal').modal('open');
 					});
 					$scope.currentProject = project
 				}
 
-			},
-			controllerAs: 'lc'
-		}
-	}]);
+				/* If the url has parameters, then this section is
+				executed. It checks whether url has a search query
+				or is it project url. */ 
+				var searchObject = $location.url();
+				if(searchObject){
+					searchObject = searchObject.split('=');
+					if(searchObject.length == 2 && searchObject[0].indexOf('search') ==1){
```

Comment:
you'll wanna use angular routes for the pages, e.g. `#projects` for the projects view `#mentors` for the mentors page. Then use the get parameters for any parameters on the same page, `?q=...` or `?project=...`